File: test/test-os.cc
/* ***** BEGIN LICENSE BLOCK *****
 * Distributed under the BSD license:
 *
 * Copyright (c) 2015, xuewen.chu
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *     * Redistributions of source code must retain the above copyright
 *       notice, this list of conditions and the following disclaimer.
 *     * Redistributions in binary form must reproduce the above copyright
 *       notice, this list of conditions and the following disclaimer in the
 *       documentation and/or other materials provided with the distribution.
 *     * Neither the name of xuewen.chu nor the
 *       names of its contributors may be used to endorse or promote products
 *       derived from this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL xuewen.chu BE LIABLE FOR ANY
 * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *
 * ***** END LICENSE BLOCK ***** */

#include "flare/util/os.h"
#include "flare/util/string.h"

using namespace flare;

void test_os(int argc, char **argv) {
	F_LOG(os::info());
	F_LOG(os::version());
	F_LOG(os::brand());
	F_LOG(os::subsystem());
	F_LOG(os::language());
	F_LOG(os::is_wifi());
	F_LOG(os::is_mobile());
	F_LOG(os::network_status());
	F_LOG(os::is_ac_power());
	F_LOG(os::is_battery());
	F_LOG(os::battery_level());
	F_LOG(os::memory());
	F_LOG(os::used_memory());
	F_LOG(os::available_memory());
}

File: test/testing/test_os.ts

import * as os from 'quark/os'
import { LOG,Mv } from './tool'

export default async function(_: any) {
	LOG('\nOs:\n');
	Mv(os, 'version', [])
	Mv(os, 'brand', [])
	Mv(os, 'model', [])
	Mv(os, 'info', [])
	Mv(os, 'languages', [])
	Mv(os, 'isWifi', [])
	Mv(os, 'isMobile', [])
	Mv(os, 'networkInterface', [])
	Mv(os, 'isAcPower', [])
	Mv(os, 'isBattery', [])
	Mv(os, 'batteryLevel', [])
	Mv(os, 'memory', [])
	Mv(os, 'usedMemory', [])
	Mv(os, 'availableMemory', [])
	Mv(os, 'cpuUsage', [])
}